aldryn_reversion/admin.py

```diff
@@ -100,6 +101,49 @@ def post_delete_plugin(self, request, plugin):
         comment = u'Deleted plugin #{0.id}: {0!s}'.format(plugin)
         self._create_revision(plugin, request.user, comment)
 
+    def log_addition(self, request, obj):
+        """
+        Override reversion.VersionAdmin log addition to provide useful message.
+        """
+        super(reversion.VersionAdmin, self).log_addition(request, obj)
+        comment = _("Initial version of {0}.{1}".format(
+            build_obj_repr(obj), get_translation_info_message(obj)))
+        self.revision_manager.save_revision(
+            self.get_revision_data(request, obj),
+            user=request.user,
+            comment=comment,
+            ignore_duplicates=self.ignore_duplicate_revisions,
+            db=self.revision_context_manager.get_db(),
+        )
+
+    def log_change(self, request, obj, message):
+        # prepare correct change message so that we can distinguish which
+        # revision would be restored. if object has language code - apppend
+        # it to the message
+        new_message = u"{0} {1}{2}".format(
+            message, build_obj_repr(obj), get_translation_info_message(obj))
+        super(VersionedPlaceholderAdminMixin, self).log_change(
+            request, obj, new_message)
+
+    # TODO: extract to separate translation admin mixin
+    def log_deletion(self, request, obj, object_repr):
+        # skip not translation objects, we don't need to do anything on regular
+        # objects
+        if not object_is_translation(obj):
+            super(VersionedPlaceholderAdminMixin, self).log_deletion(
+                request, obj, object_repr)
+            return
+
+        # django-reversion does not provides you with log deletion,
+        # for recover view it just uses diff between real objects
+        # and the last revision objects for given model.
+        # Instead wev will use log_change for translation master object
+        message = _(
```

Comment:
Not sure this is the best way to create the translated string. Might be best to use the old style % method here.